client/services/FeedbackGeneratorService.js

```diff
@@ -415,6 +415,20 @@ tie.factory('FeedbackGeneratorService', [
             'Please keep your code within the existing predefined functions',
             '-- we cannot process code in the global scope.'
           ].join(' '));
+        } else if (prereqCheckFailure.hasWrongLanguage()) {
+          WRONG_LANGUAGE_ERRORS.python.forEach(function(error) {
+            if (error.errorName === prereqCheckFailure.getWrongLangKey()) {
+              error.feedbackParagraphs.forEach(function(json) {
+                if (json._type === 'text') {
```

Comment:
We should really be working with actual domain objects if you have anything more complicated than, say, a string. It's a bit hard to maintain plain js objects because they are so mutable (eg keys can be added to them at any time, etc)

So, when you say "simplified representation of a FeedbackParagraph object" what exactly do you mean? IE what exactly is "json"?